velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -135,21 +135,65 @@ class S3ReadFile final : public ReadFile {
 
 namespace filesystems {
 
-namespace S3Config {
-namespace {
-constexpr char const* kPathAccessStyle{"hive.s3.path-style-access"};
-constexpr char const* kEndpoint{"hive.s3.endpoint"};
-constexpr char const* kSecretKey{"hive.s3.aws-secret-key"};
-constexpr char const* kAccessKey{"hive.s3.aws-access-key"};
-constexpr char const* kSSLEnabled{"hive.s3.ssl.enabled"};
-constexpr char const* kUseInstanceCredentials{
-    "hive.s3.use-instance-credentials"};
-} // namespace
-} // namespace S3Config
+class S3Config {
+ public:
+  S3Config(const Config* config) {
+    // Default is to use virtual addressing.
+    // Hence, path-style-access is false.
+    useVirtualAddressing_ = !config->get("hive.s3.path-style-access", false);
```

Comment:
The code might be simpler if we just call config_->get(...) from the getters and use a single member variable  config_.